pkg/kubectl/sorting_printer.go

```diff
@@ -153,11 +154,56 @@ func isLess(i, j reflect.Value) (bool, error) {
 		return i.String() < j.String(), nil
 	case reflect.Ptr:
 		return isLess(i.Elem(), j.Elem())
+	case reflect.Struct:
+		// special case handling, if the struct has methods: Before, etc
+		// TODO: add more?
+		supported, less := lessFunc(i, j, "Before")
+		if supported {
+			return less, nil
+		}
+		// fallback for the fields comparision
+		for idx := 0; idx < i.NumField(); idx++ {
+			less, err := isLess(i.Field(idx), j.Field(idx))
+			if err != nil || !less {
+				return less, err
+			}
+		}
+		return true, nil
+	case reflect.Array, reflect.Slice:
+		// note: the length of i and j may be different
+		for idx := 0; idx < integer.IntMin(i.Len(), j.Len()); idx++ {
+			less, err := isLess(i.Index(idx), j.Index(idx))
+			if err != nil || !less {
+				return less, err
+			}
+		}
+		return true, nil
 	default:
 		return false, fmt.Errorf("unsortable type: %v", i.Kind())
 	}
 }
 
+// this is a helper func to compare two structs when the given methods exists
+func lessFunc(i, j reflect.Value, methods ...string) (supported, less bool) {
+	for _, name := range methods {
+		f := i.MethodByName(name)
+		fType := f.Type()
+		// it must be a func and take a single argument which has the same type as j
+		// also it must returns a bool value
```

Comment:
This helper will end up being really clunky if it tries to handle every single case using reflection. That's why I am in favor of scoped helpers. You avoid specifying magic strings like "Before" and the code will be more maintainable eg. with three methods specified, which one maps to which case? Also timestamps are the only data represented as structs in our object meta so I am not sure we will ever need any other case.